tests/FSharp.Core.UnitTests/FSharp.Core/PrimTypes.fs

```diff
@@ -905,32 +987,100 @@ module internal RangeTestsHelpers =
 
+        Assert.AreEqual ({min0 .. max0 .. max0}, seq {yield min0; yield min0 + max0})
+        Assert.AreEqual ({min0 .. max1 .. max0}, seq {min0; min0 + max1})
+        Assert.AreEqual ({min0 .. max2 .. max0}, seq {min0; min0 + max2})
+        Assert.AreEqual ({min0 .. max3 .. max0}, seq {min0; min0 + max3})
+
         Assert.AreEqual ([min0 .. max0 .. max0], [min0; min0 + max0])
         Assert.AreEqual ([min0 .. max1 .. max0], [min0; min0 + max1])
         Assert.AreEqual ([min0 .. max2 .. max0], [min0; min0 + max2])
         Assert.AreEqual ([min0 .. max3 .. max0], [min0; min0 + max3])
 
+        Assert.AreEqual ([|min0 .. max0 .. max0|], [|min0; min0 + max0|])
+        Assert.AreEqual ([|min0 .. max1 .. max0|], [|min0; min0 + max1|])
+        Assert.AreEqual ([|min0 .. max2 .. max0|], [|min0; min0 + max2|])
+        Assert.AreEqual ([|min0 .. max3 .. max0|], [|min0; min0 + max3|])
 
-type RangeTests() =
-    [<Fact>] member _.``Range.SByte``  () = RangeTestsHelpers.signed   System.SByte.MinValue  System.SByte.MaxValue
-    [<Fact>] member _.``Range.Byte``   () = RangeTestsHelpers.unsigned System.Byte.MinValue   System.Byte.MaxValue
-    [<Fact>] member _.``Range.Int16``  () = RangeTestsHelpers.signed   System.Int16.MinValue  System.Int16.MaxValue
-    [<Fact>] member _.``Range.UInt16`` () = RangeTestsHelpers.unsigned System.UInt16.MinValue System.UInt16.MaxValue
-    [<Fact>] member _.``Range.Int32``  () = RangeTestsHelpers.signed   System.Int32.MinValue  System.Int32.MaxValue
-    [<Fact>] member _.``Range.UInt32`` () = RangeTestsHelpers.unsigned System.UInt32.MinValue System.UInt32.MaxValue
-    [<Fact>] member _.``Range.Int64``  () = RangeTestsHelpers.signed   System.Int64.MinValue  System.Int64.MaxValue
-    [<Fact>] member _.``Range.UInt64`` () = RangeTestsHelpers.unsigned System.UInt64.MinValue System.UInt64.MaxValue
+module RangeTests =
+    module BuildTime =
+        [<Fact>]
+        let ``Range.SByte``  () = RangeTestsHelpers.signed   System.SByte.MinValue  System.SByte.MaxValue
 
-    [<Fact>]
-    member _.``Range.IntPtr`` () =
-        if System.IntPtr.Size >= 4 then RangeTestsHelpers.signed (System.IntPtr System.Int32.MinValue) (System.IntPtr System.Int32.MaxValue)
-        if System.IntPtr.Size >= 8 then RangeTestsHelpers.signed (System.IntPtr System.Int64.MinValue) (System.IntPtr System.Int64.MaxValue)
-        
-    [<Fact>]
-    member _.``Range.UIntPtr`` () =
-        if System.UIntPtr.Size >= 4 then RangeTestsHelpers.unsigned (System.UIntPtr System.UInt32.MinValue) (System.UIntPtr System.UInt32.MaxValue)
-        if System.UIntPtr.Size >= 8 then RangeTestsHelpers.unsigned (System.UIntPtr System.UInt64.MinValue) (System.UIntPtr System.UInt64.MaxValue)
-        
+        [<Fact>]
+        let ``Range.Byte``   () = RangeTestsHelpers.unsigned System.Byte.MinValue   System.Byte.MaxValue
+
+        // Note: the IEnumerable<char> range iterator doesn't currently pass these tests. Should it?
```

Comment:
There's an overflow exception in the enumerator type:

```log
Message: 
  System.OverflowException : Arithmetic operation resulted in an overflow.

Stack Trace: 
  Range-Char@1015-14.Invoke(Char x, Char y)
  ProperIntegralRangeEnumerator`2.get_CanStep() line 5648
  IEnumerator.MoveNext() line 5608
  SeqModule.ToArray[T](IEnumerable`1 source) line 1033
  Range.Char() line 1015
  RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
  MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

Note that there are no tests for `char` in the version of these tests in `main` — they would fail if there were any, since the same machinery is used for list comprehensions in `main`.

Reply:
Probably not worth bothering with it in this PR but worth looking at this separately

Reply:
Yes, I agree.

Without the `seq` assertions, these tests do pass for `char` (i.e., the new code gen for list and array initialization passes).